### ISSUE TITLE
Generic Default Incompatibility in non generic subclass.

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#3491]
**Describe the bug**
The title is a bit long, but I think an example is best way to describe it,

```python
from typing import Generic, TypeVar

T = TypeVar("T")


class Foo(Generic[T]):
    def __init__(self, x: T = 1) -> None:
        ...


class Bar(Foo[int]):
    ...


class Baz(Bar):
    def __init__(self) -> None:
        super().__init__()  # Type error:  "Literal[1]" cannot be assigned to parameter "x" of type "T@Foo" in function "__init__"

class Baz2(Bar):
    def __init__(self) -> None:
        super().__init__(x=1)  # Good no error


class Bar2(Foo[int]):
    def __init__(self) -> None:
        super().__init__()  # Type error:  "Literal[1]" cannot be assigned to parameter "x" of type "T@Foo" in function "__init__"

class Bar3(Foo[int]):
  def __init__(self) -> None:
    super().__init__(x=1) # Good no error.
```

**To Reproduce**
Run pyright on code block. I see 2 type errors from using default value, but passing value explicitly makes it type check. I'd expect not passing default to have same type checking behavior.

**Expected behavior**
No errors and that Baz/Baz2 behave same way.

**VS Code extension or command-line**
pyright cli 1.1.247. Testing older versions looks like this error started from 1.1.199.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.248, which I just published. It will also be included in the next release of pylance.
